### ISSUE TITLE
Fails to compile when setting compiler launcher to ccache

### ISSUE THREAD

**Issue by fsimonis** [CoatiSoftware/Sourcetrail#1081]
Sourcetrail, fails to compile when `CMAKE_CXX_COMPILER_LAUNCHER` is set to ccache. I am using CMake version `3.18.2`.
```
[  0%] Building CXX object CMakeFiles/Sourcetrail_lib_utility.dir/src/lib_utility/utility/TextCodec.cpp.o
ccache: error: Recursive invocation (the name of the ccache binary must be "ccache")
make[2]: *** [CMakeFiles/Sourcetrail_lib_utility.dir/build.make:82: CMakeFiles/Sourcetrail_lib_utility.dir/src/lib_utility/utility/TextCodec.cpp.o] Error 1
ccache: error: Recursive invocation (the name of the ccache binary must be "ccache")

```

**Comment by mlangkabel** [comment]
I think this is caused by these lines in our CMakeLists file:

```
# speed up recompiling on unix with ccache
find_program(CCACHE_PROGRAM ccache)
if (CCACHE_PROGRAM)
	message(STATUS "ccache found")

	# Support Unix Makefiles and Ninja
	set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
endif()
```

In the `if` statement you could try to add a check that the compiler is not already `ccache`. Would that work?

**Comment by fsimonis** [comment]
You could also set [`CMAKE_CXX_COMPILER_LAUNCHER`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_LAUNCHER.html).Then CMake should do the rest internally.

This could look like this:
```cmake
if(NOT DEFINED CMAKE_CXX_COMPILER_LAUNCHER)
  find_program(CCACHE_PROGRAM ccache)
  if (CCACHE_PROGRAM)
    set(CMAKE_CXX_COMPILER_LAUNCHER "${CCACHE_PROGRAM}" CACHE STRING "Compiler launcher for CXX")
  endif()
endif()
```

**Comment by mlangkabel** [comment]
Created pull request #1093 with your suggested fix

**Comment by mlangkabel** [comment]
Fixed by merge of pull request #1093.
